beacon/beacon.go

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (C) 2015 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+// Package beacon contains an implementation of GA4GH Beacon API (http://ga4gh.org/#/beacon).
+package beacon
+
+import (
+	"encoding/xml"
+	"fmt"
+	"html/template"
+	"net/http"
+	"strconv"
+
+	"github.com/googlegenomics/beacon-go/internal/query"
+	"google.golang.org/appengine"
+)
+
+// BeaconAPI implements a GA4GH Beacon API (http://ga4gh.org/#/beacon) backed
+// by a Google Cloud BigQuery allele table.
+type BeaconAPI struct {
```

Comment:
This is what we refer to as name stuttering

https://golang.org/doc/effective_go.html#package-names

If you name this "API" (or probably "Server" is more correct) you end up with

  beacon.Server

which is nicer than

  beacon.BeaconAPI